models/model.js

```diff
@@ -1,95 +1,152 @@
 const db = require("../db/connection");
+const { checkExists } = require("../db/seeds/utils");
 
 exports.fetchTopics = () => {  
   return db
-    .query(
-        `SELECT * FROM topics`     
-        )
+    .query(`SELECT * FROM topics`)
     .then(({ rows }) => {
-        if (rows.length == 0) {
-            return Promise.reject({status: 404, msg: "This data does not exist"})
-         }
-         else { 
-      return rows;
-         }
+      if (rows.length == 0) {
+        return Promise.reject({status: 404, msg: "This data does not exist"})
+      }
+      else { return rows;}
     })
-    
 };
 
-exports.fetchArticleByID = (article_id) => {  
-    return db
-    .query( `
-    SELECT articles.*, COUNT (comments.comment_id) AS comment_count 
+
+
+exports.fetchUsers = () => {  
+  return db
+    .query(`SELECT * FROM users`)
+    .then(({ rows }) => {
+      if (rows.length == 0) {
+          return Promise.reject({status: 404, msg: "This data does not exist"})
+      }
+      else {return rows}  
+    })
+};
+
+
+
+exports.fetchArticles = (sort_by, order) => {
+  if (sort_by == undefined) {
+    sort_by = 'created_at'
+  }
+  if (order == undefined) {
+    order = 'DESC'
+  }
+
+  if (!['votes', 'created_at','author','article_id'].includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: 'Invalid sort query' });
+  }
+  if (!['ASC', 'DESC'].includes(order)) {
+    return Promise.reject({ status: 400, msg: 'Invalid order query' });
+  }
+
+  return db
+    .query(`
+    SELECT DISTINCT articles.*, COUNT (comments.comment_id) ::INT AS comment_count 
     FROM articles
     JOIN comments ON comments.article_id = articles.article_id
-    WHERE articles.article_id = $1
-    GROUP BY articles.article_id`, [article_id]
-    
-    )
+    GROUP BY articles.article_id
+    ORDER BY ${sort_by} ${order}`)
       .then(({ rows }) => {
         if (rows.length == 0) {
-           return Promise.reject({status: 404, msg: "This article does not exist"})
+          return Promise.reject({status: 404, msg: "This article does not exist"})
```

Comment:
This was an earlier task but it looks like you haven't added the functionality to query by a topic?